ui/desktop/app/controllers/scopes/scope/projects/targets/index.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
+
+export default class ScopesScopeProjectsTargetsIndexController extends Controller {
+  // =services
+
+  @service confirm;
+  @service ipc;
+  @service router;
+  @service session;
+  @service store;
+
+  // =methods
+
+  /**
+   * Quick connect method used to call main connect method and handle
+   * connection errors unique to this route
+   * @param {TargetModel} target
+   */
+  @action
+  async quickConnect(target) {
+    try {
+      await this.connect(target);
+    } catch (error) {
+      this.confirm
+        .confirm(error.message, { isConnectError: true })
+        // Retry
+        .then(() => this.quickConnect(target));
+    }
+  }
+
+  /**
+   * Establish a session to current target.
+   * @param {TargetModel} target
+   * @param {HostModel} host
+   */
+  @action
+  @loading
+  async connect(target, host) {
```

Comment:
None blocking comment: we moved `connect` from the route file to the controller file. Can you briefly explain why? I do like more this approach tho. Makes more sense on what a controller is, but I would like to hear your thoughts.

Reply:
Yes! I actually tried keeping it in the route as that is the standard in our app. However, when trying to utilize the connect `route-action` in the child route, we lost the ability to await the connect. This way, we can properly await the connect without showing that the route has changed in the event of a successful connection.